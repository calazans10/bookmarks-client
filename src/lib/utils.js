export const isPaginationVisible = (count, limit, total) => count <= total && total > limit;
