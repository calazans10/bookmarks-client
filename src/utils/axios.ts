import axios from 'axios';

const enhanceError = error => {
  const defaultMessage = 'The operation could not be completed. Please try again later.';
  const enhancedError = error || {};

  if (!enhancedError.response) {
    enhancedError.response = {};
    enhancedError.response.message = defaultMessage;
  }

  const { status } = enhancedError.response;
  if (!status) {
    enhancedError.response.status = 0;
  }

  return enhancedError;
};

class AxiosFactory {
  createInstance(url) {
    const source = axios.CancelToken.source();
    this.instance = axios.create({
      baseURL: url,
      timeout: process.env.REACT_APP_TIMEOUT,
      cancelToken: source.token,
    });

    this.instance.interceptors.request.use(
      config => {
        const token = sessionStorage.getItem('token');
        if (token) {
          config.headers.Authorization = `Bearer ${token}`;
        }

        setTimeout(() => {
          source.cancel();
        }, process.env.REACT_APP_TIMEOUT + 1000);

        return config;
      },
      error => {
        const enhancedError = enhanceError(error);
        Promise.reject(enhancedError);
      }
    );

    this.instance.interceptors.response.use(
      config => config,
      error => {
        const enhancedError = enhanceError(error);
        return Promise.reject(enhancedError);
      }
    );

    return this.instance;
  }
}

export default new AxiosFactory();
