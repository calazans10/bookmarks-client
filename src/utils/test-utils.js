import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import rootReducer from '../reducers';

export function renderWithRedux(
  ui,
  { initialState, store = createStore(rootReducer, initialState) } = {}
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}
