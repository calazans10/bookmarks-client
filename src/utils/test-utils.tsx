import React, { ReactNode } from 'react';
import { AnyAction, Action, createStore, Store } from 'redux';
import { Provider } from 'react-redux';
import { render, RenderResult } from '@testing-library/react';
import rootReducer from '../reducers';

interface RenderWithRedux<
  S = any,
  A extends Action = AnyAction,
  I extends S = any
  > {
  (
    ui: ReactNode,
    reduxOptions?: {
      store?: Store<S, A>
      initialState?: I
    }
  ): RenderResult & {
    store: Store<S, A>
  }
}

export const renderWithRedux: RenderWithRedux = (
  ui,
  { initialState, store = createStore(rootReducer, initialState) } = {}
) => {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}
