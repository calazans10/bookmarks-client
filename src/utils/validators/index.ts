// eslint-disable-next-line no-useless-escape
const EMAIL_REGEX = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
const PASSWORD_REGEX = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,}$/;
const NUMERIC_REGEX = /^[0-9]+$/;
const FULLNAME_REGEX = /^(.+?) ([^\s,]+)(,? (?:[JS]r\.?|III?|IV))?$/i;
const URL_REGEX = /(^|\s)((https?:\/\/)?[\w-]+(\.[\w-]+)+\.?(:\d+)?(\/\S*)?)/gi;

const required = value => (value ? undefined : 'É necessário preencher este campo.');

const mustChooseOne = value =>
  typeof value === 'undefined' || value.length === 0
    ? 'É necessário escolher ao menos uma opção.'
    : undefined;

const mustBeEmail = value => {
  if (!value || EMAIL_REGEX.test(value)) {
    return undefined;
  }

  return 'O e-mail informado não é válido.';
};

const mustBePassword = value => {
  if (!value || PASSWORD_REGEX.test(value)) {
    return undefined;
  }

  return 'Senha inválida. Use oito ou mais caracteres contendo no mínimo, uma letra maiúscula, uma letra minúscula e um número.';
};

const mustBeNumeric = value => {
  if (!value || NUMERIC_REGEX.test(value.replace(/\W/g, ''))) {
    return undefined;
  }

  return 'Este campo só aceita números.';
};

const mustBeUrl = value => {
  if (!value || URL_REGEX.test(value)) {
    return undefined;
  }

  return 'Digite um endereço válido.';
};

const mustBeFullName = value => {
  if (!value || FULLNAME_REGEX.test(value)) {
    return undefined;
  }

  return 'Digite o nome completo.';
};

const minLength = min => value => {
  if (!value || value.replace(/\W/g, '').length >= min) {
    return undefined;
  }

  return `Este campo requer no mínimo ${min} ${min === 1 ? 'caractere' : 'caracteres'}.`;
};

const maxLength = max => value => {
  if (!value || value.replace(/\W/g, '').length <= max) {
    return undefined;
  }

  return `Este campo requer no máximo ${max} ${max === 1 ? 'caractere' : 'caracteres'}.`;
};

const composeValidators = (...validators) => value =>
  validators.reduce((error, validator) => error || validator(value), undefined);

export {
  required,
  mustChooseOne,
  mustBeEmail,
  mustBePassword,
  mustBeNumeric,
  mustBeUrl,
  mustBeFullName,
  minLength,
  maxLength,
  composeValidators,
};
