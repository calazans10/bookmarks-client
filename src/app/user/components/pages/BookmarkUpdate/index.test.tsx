import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { bookmarks, users } from 'fixtures';
import { renderWithRedux } from 'utils/test-utils';
import { BookmarkUpdate } from './';

describe('BookmarkUpdate', () => {
  it('renders without crashing', () => {
    const user = users.find(user => !user.is_admin);
    const filteredBookmarks = bookmarks.filter(bookmark => bookmark.user_id === user!.id);

    const { container } = renderWithRedux(
      <MemoryRouter>
        <BookmarkUpdate
          bookmark={filteredBookmarks[0]}
          onRequestUpdateBookmark={jest.fn()}
        />
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
