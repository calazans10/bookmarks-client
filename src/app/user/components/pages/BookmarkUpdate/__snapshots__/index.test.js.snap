// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkUpdate renders without crashing 1`] = `
<div
  class="sc-fzXfLS rWSAT"
>
  <div
    class="sc-fzXfLO eNxDZJ"
  >
    <p
      class="sc-fzXfLP kvHAaA"
    >
      <span
        class="sc-fzXfLQ ehFqOr"
      />
       
      
    </p>
    <button
      aria-label="Close alert"
      class="sc-fzXfLR iJddGT"
      type="button"
    >
      <span>
        Close
      </span>
    </button>
  </div>
  <header
    class="sc-fzXfLT dkBtqJ"
  >
    <div
      class="sc-fzXfLU biEnOK"
    >
      <div
        class="sc-AykKJ yuoou"
      >
        <a
          class="sc-AykKK BJsjk"
          href="/bookmarks"
        >
          Bookmarks
        </a>
      </div>
      <nav
        class="sc-fzXfLV kVucXT"
      >
        <ul>
          <li
            class="sc-fzXfLW czKXzQ"
          >
            <button
              class="sc-fzXfLX hlqnDY"
              type="button"
            >
              Sign Out
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </header>
  <main
    class="sc-fzXfMv hgkbuk"
  >
    <form
      autocomplete="off"
      class="sc-AykKC hOQMpN"
      novalidate=""
    >
      <fieldset>
        <legend
          class="sc-AykKD jWsEOj"
        >
          Edit bookmark
        </legend>
        <div
          class="sc-AykKE dMuQNO"
        >
          <label
            class="sc-AykKF fboVCh"
            for="title"
          >
            Title
          </label>
          <input
            class="sc-AykKG glNtFx"
            id="title"
            name="title"
            placeholder="Title"
            type="text"
            value="Introducing the React RFC Process"
          />
        </div>
        <div
          class="sc-AykKE dMuQNO"
        >
          <label
            class="sc-AykKF fboVCh"
            for="url"
          >
            Url
          </label>
          <input
            class="sc-AykKG glNtFx"
            id="url"
            name="url"
            placeholder="Url"
            type="url"
            value="https://reactjs.org/blog/2017/12/07/introducing-the-react-rfc-process.html"
          />
        </div>
        <div>
          <button
            class="sc-AykKI bmvfEn"
            type="submit"
          >
            Update
          </button>
          <a
            href="/bookmarks"
          >
            Cancel
          </a>
        </div>
      </fieldset>
    </form>
  </main>
</div>
`;
